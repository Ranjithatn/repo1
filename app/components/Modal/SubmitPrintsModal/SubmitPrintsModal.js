import React, { Fragment } from "react";
import { connect } from "react-redux";

import Modal from "../Modal";
import Title from "../../Title/Title";
import Button from "../../Button/Button";
import Switch from "../../Switch/Switch";
import { requestHideModal } from "../../../actions/modal";
import { requestInputFieldChanged } from "../../../actions/global";
import { requestSubmitPrints } from "../../../actions/jobs";
import "./SubmitPrintsModal.scss";

export const SubmitPrintsModal = ({
  requestHideModal,
  formatMessage,
  selectedSearchOption,
  requestInputFieldChanged,
  requestSubmitPrints
}) => {
  return (
    <Modal
      title={formatMessage({ id: "submitPrints" })}
      className="submitPrints-modal"
      requestHideModal={requestHideModal}
      content={
        <Fragment>
          <Title
            is="5"
            text={formatMessage({ id: "newAction.searchOptions" })}
          />
          <Switch
            id="submitPrints-crminalDatabase"
            onChange={e => requestInputFieldChanged(e, "submitPrints")}
            value="Search Criminal Database"
            text={formatMessage({ id: "searchCriminalDB" })}
            className="is-primary is-rounded"
            checked={selectedSearchOption === "Search Criminal Database"}
          />
          <Switch
            id="submitPrints-civilDatabase"
            onChange={e => requestInputFieldChanged(e, "submitPrints")}
            value="Search Latent Database"
            text={formatMessage({ id: "searchLatentDB" })}
            className="is-primary is-rounded"
            checked={selectedSearchOption === "Search Latent Database"}
          />
          <Switch
            id="submitPrints-latentDatabase"
            onChange={e => requestInputFieldChanged(e, "submitPrints")}
            value="Search Civil Database"
            text={formatMessage({ id: "searchCivilDB" })}
            className="is-primary is-rounded"
            checked={selectedSearchOption === "Search Civil Database"}
          />
        </Fragment>
      }
      buttons={
        <Fragment>
          <Button
            id="submitPrints-btn"
            className="is-primary"
            text={formatMessage({id: "Submit"})}
            disabled={!selectedSearchOption}
            onClick={() => requestSubmitPrints()}
          />
        </Fragment>
      }
    />
  );
};

const mapState = state => ({
  selectedSearchOption: state.jobs.newJob.selectedSearchOption
});

export default connect(mapState, {
  requestHideModal,
  requestInputFieldChanged,
  requestSubmitPrints
})(SubmitPrintsModal);
