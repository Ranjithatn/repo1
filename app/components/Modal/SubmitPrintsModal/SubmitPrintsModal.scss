.submitPrints-modal footer.modal-card-foot {
  display: flex;
  justify-content: flex-end;
}