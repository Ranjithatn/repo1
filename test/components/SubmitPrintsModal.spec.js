import React from "react";
import toJson from "enzyme-to-json";
import { SubmitPrintsModal } from "../../app/components/Modal/SubmitPrintsModal/SubmitPrintsModal";
import { shallow, configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

let component, tree, instance;
let criminalDatabaseToggle, civilDatabaseToggle, latentDatabaseToggle;

let props = {
  requestHideModal: jest.fn(),
  requestInputFieldChanged: jest.fn(),
  selectedSearchOption: "Search Criminal Database",
  formatMessage: jest.fn(),
  requestSubmitPrints: jest.fn()
}

beforeEach(() => {
  component = mount(
    <SubmitPrintsModal {...props} />
  );
  criminalDatabaseToggle = component.find("input#submitPrints-crminalDatabase");
  civilDatabaseToggle = component.find("input#submitPrints-civilDatabase");
  latentDatabaseToggle = component.find("input#submitPrints-latentDatabase");
  tree = toJson(component);
  instance = component.instance();
});

describe("<SubmitPrintsModal />", () => {
  it("should render without crashing", () => {
    expect(tree).toMatchSnapshot();
  });
  it('should have a criminal database toggle option', () => {
    const output = component.find("Switch#submitPrints-crminalDatabase").length;
    expect(output).toEqual(1);
  })
  it('should have a civil database toggle option', () => {
    const output = component.find("Switch#submitPrints-civilDatabase").length;
    expect(output).toEqual(1);
  })
  it('should have a latent database toggle option', () => {
    const output = component.find("Switch#submitPrints-latentDatabase").length;
    expect(output).toEqual(1);
  })
  it("should call requestInputFieldChanged when criminal database is toggled", () => {
    let event = {
      id: "submitPrints-crminalDatabase"
    };
    criminalDatabaseToggle.simulate("change", event);
    expect(props.requestInputFieldChanged).toHaveBeenCalled();
  });
  it("should call requestInputFieldChanged when civil database is toggled", () => {
     let event = {
      id: "submitPrints-civilDatabase"
    };
    civilDatabaseToggle.simulate("change", event);
    expect(props.requestInputFieldChanged).toHaveBeenCalled();
  });
  it("should call requestInputFieldChanged when latent database is toggled", () => {
    let event = {
      id: "submitPrints-latentDatabase"
    };
    latentDatabaseToggle.simulate("change", event);
    expect(props.requestInputFieldChanged).toHaveBeenCalled();
  });
  it("should call requestSubmitPrints when submit button is clicked", () => {
    const submitBtn = component.find('button#submitPrints-btn').simulate('click', { target: { id: 'submitClicked'}});
    expect(props.requestSubmitPrints).toHaveBeenCalled();
  });
});
