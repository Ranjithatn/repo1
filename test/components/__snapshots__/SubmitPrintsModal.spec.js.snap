// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitPrintsModal /> should render without crashing 1`] = `
<Component
  formatMessage={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "id": "submitPrints",
          },
        ],
        Array [
          Object {
            "id": "newAction.searchOptions",
          },
        ],
        Array [
          Object {
            "id": "searchCriminalDB",
          },
        ],
        Array [
          Object {
            "id": "searchLatentDB",
          },
        ],
        Array [
          Object {
            "id": "searchCivilDB",
          },
        ],
        Array [
          Object {
            "id": "Submit",
          },
        ],
      ],
    }
  }
  requestHideModal={[MockFunction]}
  requestInputFieldChanged={[MockFunction]}
  requestSubmitPrints={[MockFunction]}
  selectedSearchOption="Search Criminal Database"
>
  <Modal
    buttons={
      <UNDEFINED>
        <Button
          className="is-primary"
          disabled={false}
          id="submitPrints-btn"
          onClick={[Function]}
          text={undefined}
        />
      </UNDEFINED>
    }
    className="submitPrints-modal"
    content={
      <UNDEFINED>
        <Title
          as="h1"
          is="5"
          text={undefined}
        />
        <Switch
          checked={true}
          className="is-primary is-rounded"
          id="submitPrints-crminalDatabase"
          onChange={[Function]}
          text={undefined}
          value="Search Criminal Database"
        />
        <Switch
          checked={false}
          className="is-primary is-rounded"
          id="submitPrints-civilDatabase"
          onChange={[Function]}
          text={undefined}
          value="Search Latent Database"
        />
        <Switch
          checked={false}
          className="is-primary is-rounded"
          id="submitPrints-latentDatabase"
          onChange={[Function]}
          text={undefined}
          value="Search Civil Database"
        />
      </UNDEFINED>
    }
    requestHideModal={[MockFunction]}
  >
    <div
      className="modal is-active submitPrints-modal"
      style={
        Object {
          "zIndex": "50",
        }
      }
    >
      <div
        className="modal-background"
      />
      <div
        className="modal-card"
      >
        <header
          className="modal-card-head"
        >
          <p
            className="modal-card-title"
          />
          <button
            aria-label="close"
            className="delete is-large"
            name="btn-modal-close"
            onClick={[MockFunction]}
          />
        </header>
        <section
          className="modal-card-body"
        >
          <Title
            as="h1"
            is="5"
          >
            <h1
              className="title is-5"
            />
          </Title>
          <Switch
            checked={true}
            className="is-primary is-rounded"
            id="submitPrints-crminalDatabase"
            onChange={[Function]}
            value="Search Criminal Database"
          >
            <div
              className="field"
            >
              <input
                checked={true}
                className="switch is-primary is-rounded"
                disabled={false}
                id="submitPrints-crminalDatabase"
                name="submitPrints-crminalDatabase"
                onChange={[Function]}
                type="checkbox"
                value="Search Criminal Database"
              />
              <label
                htmlFor="submitPrints-crminalDatabase"
              />
            </div>
          </Switch>
          <Switch
            checked={false}
            className="is-primary is-rounded"
            id="submitPrints-civilDatabase"
            onChange={[Function]}
            value="Search Latent Database"
          >
            <div
              className="field"
            >
              <input
                checked={false}
                className="switch is-primary is-rounded"
                disabled={false}
                id="submitPrints-civilDatabase"
                name="submitPrints-civilDatabase"
                onChange={[Function]}
                type="checkbox"
                value="Search Latent Database"
              />
              <label
                htmlFor="submitPrints-civilDatabase"
              />
            </div>
          </Switch>
          <Switch
            checked={false}
            className="is-primary is-rounded"
            id="submitPrints-latentDatabase"
            onChange={[Function]}
            value="Search Civil Database"
          >
            <div
              className="field"
            >
              <input
                checked={false}
                className="switch is-primary is-rounded"
                disabled={false}
                id="submitPrints-latentDatabase"
                name="submitPrints-latentDatabase"
                onChange={[Function]}
                type="checkbox"
                value="Search Civil Database"
              />
              <label
                htmlFor="submitPrints-latentDatabase"
              />
            </div>
          </Switch>
        </section>
        <footer
          className="modal-card-foot"
        >
          <Button
            className="is-primary"
            disabled={false}
            id="submitPrints-btn"
            onClick={[Function]}
          >
            <button
              className="button is-primary"
              disabled={false}
              id="submitPrints-btn"
              onClick={[Function]}
              style={Object {}}
              type="button"
            />
          </Button>
        </footer>
      </div>
    </div>
  </Modal>
</Component>
`;
